lib/services/nodes-api-service.js

```diff
@@ -451,5 +453,80 @@ function nodeApiServiceFactory(
             });
     };
 
+    /**
+     * Add the tags to the specified node
+     * @param  {String}     id
+     * @param  {Array}      tags
+     * @return {Promise}
+     */
+    NodeApiService.prototype.addTagsById = function(id, tags) {
+        return Promise.resolve().then(function() {
+            assert.ok(Array.isArray(tags), 'tags must be an array');
+            assert.isMongoId(id, 'the id must be a valid mongo id');
+        })
+        .then(function() {
+            return waterline.nodes.needByIdentifier(id);
+        })
+        .then(function () {
+            return waterline.nodes.addTags(id, tags);
+        })
+        .then(function() {
+            return tags;
+        });
+    };
+
+    /**
+     * Remove the tag from the specified node
+     * @param  {String}     id
+     * @param  {String}     tagName
+     * @return {Promise}
+     */
+    NodeApiService.prototype.removeTagsById = function(id, tagName) {
+        return Promise.resolve().then(function() {
+            assert.string(tagName, 'tag must be a string');
+            assert.isMongoId(id, 'the id must be a valid mongo id');
+        })
+        .then(function() {
+            return waterline.nodes.needByIdentifier(id);
+        })
+        .then(function () {
+            return waterline.nodes.remTags(id, tagName);
+        })
+        .then(function() {
+            return tagName;
+        });
+    };
+
+    /**
+     * Get a list of tags applied to the specified id
+     * @param  {String}     id
+     * @return {Promise}    Resolves to an array of tags
+     */
+    NodeApiService.prototype.getTagsById = function(id) {
+        return Promise.resolve().then(function() {
```

Comment:
@zyoung51 I believe these assertion failures will turn into HTTP 400 responses.  I'm not sure if that's what you intended, but it seems like we might want to return HTTP 500 if any of these asserts fail.